torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -518,7 +524,7 @@ std::vector<ExprHandle> TensorExprKernel::sizesForValue(
   // need to infer it.
   if (v->type()->kind() == TypeKind::TensorType) {
     auto tt = v->type()->cast<TensorType>();
-    if (tt->isComplete()) {
+    if (tt->sizes().isComplete()) {
```

Comment:
Shouldn't we do `tt->sizes().concrete_sizes()` here instead? It looks like `isComplete()` is checking for more than just if the dims in sizes are not null. See https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L445

Reply:
Correct, but that's intentional in this PR and the justification is in the commit message :)

Reply:
I'm not talking about `tt->isComplete()`. I understand that is not the right thing to do here.

I am referring to `tt->sizes().isComplete()` which is the update you have made. In `VaryingShape`, `concrete_sizes()` only checks if the dims in the shape are not null, which is what we want here. On the other hand, `isComplete()` on `VaryingShape` checks for more, which seems unnecessary here.